xonsh/main.py

```diff
@@ -233,7 +230,7 @@ def main(argv=None):
         code = sys.stdin.read()
         run_code_with_cache(code, shell.execer, glb=shell.ctx, loc=None,
                             mode='exec')
-    else:
+    elif len(sys.argv) == 1:
```

Comment:
yes, that is just for precaution allthough this clause could actually go on the top of main with this condition cause the check is easy and this case is the most frequently used i suppose.